dds/idl/be_global.cpp

```diff
@@ -571,7 +579,48 @@ BE_GlobalData::cache_topic_annotations()
         root->lookup_by_name("::@topic"));
     key_annotation_ = AST_Annotation_Decl::narrow_from_decl(
         root->lookup_by_name("::@key"));
+    nested_annotation_ = AST_Annotation_Decl::narrow_from_decl(
+        root->lookup_by_name("::@nested"));
+    default_nested_annotation_ = AST_Annotation_Decl::narrow_from_decl(
+        root->lookup_by_name("::@default_nested"));
+ }
+}
+
+/**
+ * @brief determines if we should treat a given node as a topic type. This could
+ * mean that it is explicitly a topic type or that @default_nested(FALSE) or
+ * --default_nested = false
+ * @param node the incomming node
+ * @return if type support should be generated
+ * @see BE_GlobalData::is_topic_type(AST_Decl* node)
+ * @see BE_GlobalData::is_default_nested(AST_Decl* node)
+ * @author ceneblock
+ */
+bool BE_GlobalData::treat_as_topic(AST_Decl *node)
+{
+  return is_topic_type(node) || !is_default_nested(node) || !is_nested_type(node);
+}
+
+/**
+ * @brief determins if a given node is nested by default or not. This may be set
+ * by the module level of @default_nesed or through the --default_nested=
+ * @param node the node to be inspected
+ * @return if the node is default nested or not
+ * @note false means it should be considered nested.
+ * @author ceneblock
+ */
+bool
+BE_GlobalData::is_default_nested(AST_Decl* node)
+{
+   AST_Annotation_Appl *default_nested_apply = dynamic_cast<AST_Decl *>(node -> defined_in()) -> annotations().find("::@default_nested");
+
+  if(default_nested_apply)
+  {
+    bool default_nested_apply_value = AST_Annotation_Member::narrow_from_decl ((*default_nested_apply)["value"]) -> value ()->ev ()->u.bval;
+    return default_nested_apply_value;
   }
+
+  return default_nested_;
```

Comment:
This method also needs to check the parent modules of the module where this node is defined.

Reply:
This would be done by recursively calling `dynamic_cast<AST_Decl *>(node -> defined_in()))`?